packages/jest-core/src/testSchedulerHelper.ts

```diff
@@ -5,17 +5,22 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import {Config} from '@jest/types';
 import {Test} from 'jest-runner';
 
 const SLOW_TEST_TIME = 1000;
 
 export function shouldRunInBand(
   tests: Array<Test>,
-  isWatchMode: boolean,
-  maxWorkers: number,
   timings: Array<number>,
+  {detectOpenHandles, maxWorkers, watch, watchAll}: Config.GlobalConfig,
 ) {
-  /**
+  // detectOpenHandles makes no sense without runInBand, because it cannot detect leaks in workers
```

Comment:
> because it cannot detect leaks in workers

Is this a limitation of `async_hooks` API? 

Anyway, since `detectOpenHandles` flag is handled before initializing TestScheduler, how about moving this logic there and adding a proper error message so users are aware? We need to alarm users about that anyway. 

I'm good passing whole config to `shouldRunInBand` 👍

Reply:
> Is this a limitation of `async_hooks` API?

I mean, you could certainly make leak detection work in each worker, but I don't think it's worth it - a `detectOpenHandles` debug run doesn't need performance through parallelization.

> adding a proper error message so users are aware? We need to alarm users about that anyway.

Not sure I understand correctly, do you want to error if `detectOpenHandles` without `runInBand` is used instead of just activating `runInBand` implicitly? I think forcing users to specify both would be quite annoying. Printing "did not exit within 1 second, please use --detectOpenHandles --runInBand" also doesn't look good :/

Reply:
On the second thought, we can hide it from the user. Would be nice to add a description note that adding this flag will make tests running serially

Reply:
Added to docs & yargs 👍